scte35/splicecommandmodify.go

```diff
@@ -0,0 +1,250 @@
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+*/
+
+package scte35
+
+import (
+	"github.com/Comcast/gots"
+)
+
+// CreateSpliceInsertCommand will create a default SpliceInsertCommand.
+func CreateSpliceInsertCommand() SpliceCommand {
+	return &spliceInsert{
+		// enable this to have less variable sized fields by
+		// default, do not use component splice mode by default
+		isProgramSplice: true,
+	}
+}
+
+// CreateTimeSignalCommand will create a default TimeSignalCommand
+func CreateTimeSignalCommand() SpliceCommand {
+	return &timeSignal{}
+}
+
+// CreateSpliceNull will create a Null SpliceCommand
+func CreateSpliceNull() SpliceCommand {
+	return &spliceNull{}
+}
+
+// SetComponentTag sets the component tag, which is used for the identification of the component.
+func (c *component) SetComponentTag(value byte) {
+	c.componentTag = value
+}
+
+// SetHasPTS sets a flag that determines if the component has a PTS.
+func (c *component) SetHasPTS(value bool) {
+	c.hasPts = value
+}
+
+// SetPTS sets the PTS of the component.
+func (c *component) SetPTS(value gots.PTS) {
+	c.pts = value & 0x01ffffffff
+}
+
+// returns the bytes of this splice command.
+func (c *spliceNull) Data() []byte {
+	return []byte{} // return empty slice
+}
+
+// SetHasPTS sets the flag that indicates if there is a pts time on the command.
+func (c *spliceNull) SetHasPTS(value bool) {
+	// do nothing
+}
+
+// SetPTS sets the PTS.
+func (c *spliceNull) SetPTS(value gots.PTS) {
+	// do nothing
+}
+
+// spliceTimeBytes returns the raw data bytes of a splice time.
+func spliceTimeBytes(hasPTS bool, pts gots.PTS) []byte {
+	if hasPTS {
+		bytes := make([]byte, 5)
+		bytes[0] = 0xFE                  // 1111 1110
+		bytes[0] |= byte(pts>>32) & 0x01 // 0000 0001
+		bytes[1] = byte(pts >> 24)       // 1111 1111
+		bytes[2] = byte(pts >> 16)       // 1111 1111
+		bytes[3] = byte(pts >> 8)        // 1111 1111
+		bytes[4] = byte(pts)             // 1111 1111
+		return bytes
+	}
+	// return 0111 1110
+	return []byte{0x7E} // only reserved bits are set
+}
+
+// returns the bytes of this splice command.
+func (c *timeSignal) Data() []byte {
```

Comment:
This and the `func (c *spliceInsert) Data() []byte` below need function names for godoc style comments